fedn/network/api/v1/model_routes.py

```diff
@@ -352,6 +352,128 @@
+    responses:
+        200:
+            description: The updated model
+            schema:
+                $ref: '#/definitions/Model'
+        404:
+            description: The model was not found
+            schema:
+                type: object
+                properties:
+                    message:
+                        type: string
+        500:
+            description: An error occurred
+            schema:
+                type: object
+                properties:
+                    message:
+                        type: string
+    """
+    try:
+        model = model_store.get(id, use_typing=False)
+
+        data = request.get_json()
+        _id = model["id"]
+
+        # Update the model with the new data
+        # Only update the fields that are present in the request
+        for key, value in data.items():
+            if key in ["_id", "model"]:
+                continue
+            model[key] = value
+
+        success = model_store.update(_id, model)
+
+        if success:
+            response = model
+            return jsonify(response), 200
+
+        return jsonify({"message": "Failed to update model"}), 500
+    except EntityNotFound as e:
+        return jsonify({"message": str(e)}), 404
+    except Exception as e:
+        return jsonify({"message": str(e)}), 500
+
+
+@bp.route("/<string:id>", methods=["PUT"])
+@jwt_auth_required(role="admin")
+def put_model(id: str):
+    """Patch model
+    Updates a model based on the provided id. All fields will be updated with the new data.
+    ---
+    tags:
+        - Models
+    parameters:
+        - name: id
+            in: path
+            required: true
+            type: string
+            description: The id or model property of the model
+        - name: model
+            in: body
+            required: true
+            type: object
+            description: The model data to update
+    responses:
+        200:
+            description: The updated model
+            schema:
+                $ref: '#/definitions/Model'
+        404:
+            description: The model was not found
+            schema:
+                type: object
+                properties:
+                    message:
+                        type: string
+        500:
+            description: An error occurred
+            schema:
+                type: object
+                properties:
+                    message:
+                        type: string
+    """
+    try:
+        model = model_store.get(id, use_typing=False)
+        data = request.get_json()
+        _id = model["id"]
+
+        success = model_store.update(_id, data)
+
+        if success:
+            response = model
+            return jsonify(response), 200
+
+        return jsonify({"message": "Failed to update model"}), 500
+    except EntityNotFound as e:
+        return jsonify({"message": str(e)}), 404
+    except Exception as e:
+        return jsonify({"message": str(e)}), 500
```

Comment:
I don't know what you guys think about these... I left it as is. Don't believe it will can cause a problem for us tbh and it's a quite convenient solution.

Reply:
it should be addressed. I.e send only specific error messages to client. We can also log (use logger and traceback) the error on server side.

Reply:
The errors returned here could be either mongo specific or if there is and EntityNotFound error it is raised on purpose.

Reply:
I think this would be more important if we were using a SQL db for example... But sure, we can change it...